src/main/java/io/neonbee/internal/handler/DefaultErrorHandler.java

```diff
@@ -35,31 +38,35 @@
     /**
      * Cached template for rendering the HTML errors.
      */
-    private final String errorTemplate;
+    private String errorTemplate;
 
     /**
-     * Returns a new ErrorHandler with the default {@link #DEFAULT_ERROR_HANDLER_TEMPLATE template}.
-     */
-    public DefaultErrorHandler() {
-        try {
-            this.errorTemplate = readErrorTemplate(DEFAULT_ERROR_HANDLER_TEMPLATE);
-        } catch (IOException e) {
-            throw new RuntimeException("Could not read default error template", e);
-        }
-    }
-
-    /**
-     * Returns a new ErrorHandler with the passed template.
+     * Initializes the ErrorHandler with the passed template.
      *
+     * @param neonbee           the related NeonBee instance
      * @param errorTemplateName resource path to the template.
-     * @throws IOException in case the template is not found or cannot be read from the current class loader.
+     *
+     * @return A Future with the initialized ErrorHandler
      */
-    public DefaultErrorHandler(String errorTemplateName) throws IOException {
-        this.errorTemplate = readErrorTemplate(errorTemplateName);
+    @Override
+    public Future<ErrorHandler> initialize(NeonBee neonbee, String errorTemplateName) {
+        return AsyncHelper.<String>executeBlocking(neonbee.getVertx(), p -> {
```

Comment:
No blocking read is required any longer, please use a non blocking read here

Reply:
We don't have an async BufferHelper.readResourceToBuffer. Should I create one? But then I would wrap if with AsynHelper again because there is no async offering of Classloader.getResourceAsStream